src/python/library/tritonclient/utils/shared_memory/shared_memory.cc

```diff
@@ -145,6 +145,8 @@ SharedMemoryRegionDestroy(void* shm_handle)
     return -5;
   }
 
+  delete handle;
```

Comment:
given the return statement above, this may not be enough.

Reply:
Good catch J! No, in this case `handle` refers to a dynamically allocated instance of our internal [struct](https://github.com/triton-inference-server/client/blob/main/src/python/library/tritonclient/utils/shared_memory/shared_memory_handle.h#L33-L44) definition as opposed to an OS handle. We are doing best-effort cleanup with `munmap` and `shm_unlink`, but we have full control to deallocate `handle`.

Testing various cases with the new change also reminded me of a minor logical error within the python client code that we should clean up here as well. Note, that I elected to use a local smart pointer to do cleanup to avoid having to write three proximal `delete`s, however, I can switch to using an explicit `delete` to match the allocation syntax.